packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1742,46 +1440,599 @@ class RenderViewport2 extends RenderBox with ContainerRenderObjectMixin<RenderSl
   void debugFillDescription(List<String> description) {
     super.debugFillDescription(description);
     description.add('$axisDirection');
-    if (_shrinkWrap)
-      description.add('shrink-wrap enabled');
-    description.add('anchor: $anchor');
     description.add('offset: $offset');
   }
 
   @override
   String debugDescribeChildren(String prefix) {
+    if (firstChild == null)
+      return '$prefix\n';
+    int count = indexOfFirstChild();
     String result = '$prefix \u2502\n';
+    RenderSliver child = firstChild;
+    while (child != lastChild) {
+      result += '${child.toStringDeep("$prefix \u251C\u2500${labelChild(count)}: ", "$prefix \u2502")}';
+      count += 1;
+      child = childAfter(child);
+    }
+    assert(child == lastChild);
+    result += '${child.toStringDeep("$prefix \u2514\u2500${labelChild(count)}: ", "$prefix  ")}';
+    return result;
+  }
+
+  // API TO BE IMPLEMENTED BY SUBCLASSES
+
+  // setupParentData
+
+  // performLayout (and optionally sizedByParent and performResize)
+
+  @protected
+  bool get hasVisualOverflow;
+
+  @protected
+  void updateOutOfBoundsData(GrowthDirection growthDirection, SliverGeometry childLayoutGeometry);
+
+  @protected
+  void updateChildPaintOffset(RenderSliver child, double paintOffset, GrowthDirection growthDirection);
+
+  @protected
+  Offset paintOffsetOf(RenderSliver child);
+
+  // applyPaintTransform
+
+  /// Returns the distance from the leading edge of the given child (as given by
+  /// the child's [SliverConstraints.axisDirection] and
+  /// [SliverConstraints.growthDirection]) to the line that crosses the child
+  /// perpendicular to the child's axis.
+  ///
+  /// The `parentMainAxisPosition` is the distance from the top (for vertical
+  /// viewports) or left (for horizontal viewports) of the viewport bounds, not
+  /// from the leading edge of the viewport.
+  @protected
+  double computeChildMainAxisPosition(RenderSliver child, double parentMainAxisPosition);
```

Comment:
Looking at the call site, I see that this function is converting the given `parentMainAxisPosition` into the coordinate system of the child.